.github/workflows/check-in-tree-build.yml

```diff
@@ -0,0 +1,175 @@
+      - '**.md'
+  pull_request:
+    branches:
+      - master
+      - llvm_release_*
+    paths-ignore: # no need to check build for documentation changes
+      - 'docs/**'
+      - '**.md'
+  schedule:
+    # Ideally, we might want to simplify our regular nightly build as we
+    # probably don't need every configuration to be built every day: most of
+    # them are only necessary in pre-commits to avoid breakages
+    - cron: 0 0 * * *
+
+env:
+  LLVM_VERSION: 12
+
+jobs:
+  build_and_test_linux:
+    name: Linux
+    strategy:
+      matrix:
+        build_type: [Release, Debug]
+        shared_libs: [NoSharedLibs]
+        include:
+          - build_type: Release
+            shared_libs: EnableSharedLibs
+      fail-fast: false
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Install dependencies
+        run: |
+          curl -L "https://apt.llvm.org/llvm-snapshot.gpg.key" | sudo apt-key add -
+          curl -L "https://packages.lunarg.com/lunarg-signing-key-pub.asc" | sudo apt-key add -
+          echo "deb https://apt.llvm.org/bionic/ llvm-toolchain-bionic main" | sudo tee -a /etc/apt/sources.list
+          echo "deb https://packages.lunarg.com/vulkan bionic main" | sudo tee -a /etc/apt/sources.list
+          sudo apt-get update
+          # Linux systems in GitHub Actions already has pre-installed clang is
+          # not needed for the translator as we rely on newer version
+          sudo apt-get --purge remove clang-8 clang-9
+          sudo apt-get -yq --no-install-suggests --no-install-recommends install \
+            clang-${{ env.LLVM_VERSION }} \
+            spirv-tools
+          sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 1000
+      - name: Checkout LLVM sources
+        uses: actions/checkout@v2
+        with:
+          repository: llvm/llvm-project
+          ref: master
+          path: llvm-project
+      - name:  Checkout the translator sources
+        uses: actions/checkout@v2
+        with:
+          path: llvm-project/llvm/projects/SPIRV-LLVM-Translator
+      - name: Configure
+        run: |
+          mkdir build && cd build
+          # ON/OFF specifically weren't used as a values for shared_libs matrix
+          # field to improve usability of PR page: instead of (Release, ON) a
+          # job will be displayed as (Release, EnableSharedLibs)
+          SHARED_LIBS=OFF
+          if [[ "${{ matrix.shared_libs }}" == "EnableSharedLibs" ]]; then
+            SHARED_LIBS=ON
+          fi
+          cmake ${{ github.workspace }}/llvm-project/llvm \
+            -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+            -DBUILD_SHARED_LIBS=${SHARED_LIBS} \
+            -DLLVM_TARGETS_TO_BUILD="X86" \
+            -DSPIRV_SKIP_CLANG_BUILD=ON \
+            -DSPIRV_SKIP_DEBUG_INFO_TESTS=ON \
+            -DLLVM_LIT_ARGS="-sv --no-progress-bar" \
+            -G "Unix Makefiles"
+      - name: Build
+        run: |
+          cd build
+          make llvm-spirv -j2
```

Comment:
I think we can skip this step. Build should be done automatically for the `check-llvm-spirv` target, right?

Reply:
> Build should be done automatically for the `check-llvm-spirv` target, right?

Yes, it would be done. However, I prefer to leave it as-is, because it allows you to understand whether CI job is failed because the project isn't buildable or there is some problem with tests. Note: in Travis CI config we also have two separate targets for build and tests